src/sentry/migrations/0505_delete_pending_deletion_rules.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 2.2.28 on 2023-07-06 22:52
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def delete_rules(apps, schema_editor):
+    from sentry.models.rule import RuleStatus
+
+    Rule = apps.get_model("sentry", "Rule")
+    for rule in RangeQuerySetWrapperWithProgressBar(
+        Rule.objects.filter(status__in=[RuleStatus.PENDING_DELETION, RuleStatus.INACTIVE])
+    ):
```

Comment:
Since there are so many rules to delete, and actually these are something like 90% of the table, I think that we're better off iterating through `Rule.objects.all` here and just filtering status in the code. 

Since `RangeQuerySetWrapper` enforces a sort and filter on `id`, and we're also filtering on `status`, there's no appropriate index to use since we don't have an index on `id, status`. So it tends to be more efficient to just walk the the whole table.